### PR TITLE
Stats

### PR DESCRIPTION
**Pull request by dwagg**

### CODE REVIEW

File: Stats.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+		return Stats.e;
+	}
+	public static final float getRt2()
+	{
+		return Stats.rt2;
+	}
+
+	//multiply n by n-1 by n-2 by ... by 2
+	public static BigInteger factorial(int n) throws IllegalArgumentException
+
```
Comment (dwagg) [inline Stats.java]: WTF extra line???

File: Stats.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+		double c = java.lang.Math.pow(1-p,n-x);
+		//System.out.println("nCx(30,6) = " + nCx + " .23^6 = " + pPowX + " pPrimePowX (1-p)^n-x = " + pPrimePowX);
+		return a * b * c;
+
+	}
+
+	//Prob there are 1,2,...,x successful events after n events where prob of success is p
+	public static double binDistCdf(int x,int n, double p) throws IllegalArgumentException{
+		double binDistCumulative = 0;
+		for (int i = 0; i <= x; i++){
```
Comment (dwagg) [inline Stats.java]: Curly brace in wrong place

File: Stats.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+		int a = Stats.choose(n,x);
+		double b = java.lang.Math.pow(p,x);
+		double c = java.lang.Math.pow(1-p,n-x);
+		//System.out.println("nCx(30,6) = " + nCx + " .23^6 = " + pPowX + " pPrimePowX (1-p)^n-x = " + pPrimePowX);
+		return a * b * c;
+
+	}
+
+	//Prob there are 1,2,...,x successful events after n events where prob of success is p
+	public static double binDistCdf(int x,int n, double p) throws IllegalArgumentException{
```
Comment (dwagg) [inline Stats.java]: Curly brace issues.

File: Stats.java [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+	//multiply n by n-1 by n-2 by ... by 2
+	public static BigInteger factorial(int n) throws IllegalArgumentException
+
+	{
+		BigInteger fact = BigInteger.ONE;
+		if (n < 0)
+		{
+			throw new IllegalArgumentException("Negative Number error.");
+		}
+		//for(int i = 0; i < n; i++)
```
Comment (dwagg) [inline Stats.java]: is this going to be used?

File: Stats.java [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+		for (int i = n; i > 1; i--)
+		{
+			fact = fact.multiply(BigInteger.valueOf(i));
+		}
+		return fact;
+	}
+
+	//In math n and k are the default arguments to the choose function
+	//how many way with n things can I choose k items
```
Comment (dwagg) [inline Stats.java]: Comment doesn't make sense. Ways?

File: Stats.java [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+			throw new IllegalArgumentException("n cannot be less than x");
+		}
+		if (n <=0 || x < 0)
+		{
+			throw new IllegalArgumentException("n must be greater than 0 and x must not be negative.");
+		}
+		int a = Stats.choose(n,x);
+		double b = java.lang.Math.pow(p,x);
+		double c = java.lang.Math.pow(1-p,n-x);
+		//System.out.println("nCx(30,6) = " + nCx + " .23^6 = " + pPowX + " pPrimePowX (1-p)^n-x = " + pPrimePowX);
```
Comment (dwagg) [inline Stats.java]: Delete debug code???

File: Stats.java [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+	//Prob there are 1,2,...,x successful events after n events where prob of success is p
+	public static double binDistCdf(int x,int n, double p) throws IllegalArgumentException{
+		double binDistCumulative = 0;
+		for (int i = 0; i <= x; i++){
+			binDistCumulative += Stats.binDistPmf(i,n,p);
+		}
+		return binDistCumulative;
+	}
+
+	public static void statsBattery()
```
Comment (dwagg) [inline Stats.java]: Add comment explaining methods

File: Stats.java [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+	{
+		if (!(p <= 1 && p >= 0))
+		{
+			throw new IllegalArgumentException("Prob not in range (0,1)");
+		}
+		if (n < x)
+		{	
+			throw new IllegalArgumentException("n cannot be less than x");
+		}
+		if (n <=0 || x < 0)
```
Comment (dwagg) [inline Stats.java]: spacing around conditionals

File: Stats.java [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+			fact = fact.multiply(BigInteger.valueOf(i));
+		}
+		return fact;
+	}
+
+	//In math n and k are the default arguments to the choose function
+	//how many way with n things can I choose k items
+	public static int choose(int n, int k)
+	{
+		BigInteger nCk = Stats.factorial(n)
```
Comment (dwagg) [inline Stats.java]: semicolon

File: Main.java [hunk 10]
```diff
@@ -3,6 +3,7 @@ public static void main(String args[])
     {
         System.out.println("running demo");
         testMathClass();
+	Stats.statsBattery();
```
Comment (dwagg) [inline Main.java]: missing space

File: Stats.java [hunk 11]
```diff
@@ -0,0 +1,88 @@
+import java.math.BigInteger;
+import java.lang.Math;
+import java.lang.IllegalArgumentException;
+public class Stats{
+	//some constats that may be used in the future
+	private static final float pi = 3.14159f;
```
Comment (dwagg) [inline Stats.java]: Make public get rid of getters

Comment (dwagg) [inline Stats.java]: Make final variable names all caps

File: Stats.java [hunk 12]
```diff
@@ -0,0 +1,88 @@
...
+	{
+		return Stats.e;
+	}
+	public static final float getRt2()
+	{
+		return Stats.rt2;
+	}
+
+	//multiply n by n-1 by n-2 by ... by 2
+	public static BigInteger factorial(int n) throws IllegalArgumentException
```
Comment (dwagg) [inline Stats.java]: why bigInts?

File: Stats.java [hunk 13]
```diff
@@ -0,0 +1,88 @@
...
+	//how many way with n things can I choose k items
+	public static int choose(int n, int k)
+	{
+		BigInteger nCk = Stats.factorial(n)
+		nCk = nCk.divide(Stats.factorial(k).multiply(factorial(n-k)));
+		return nCk.intValue();
+	}
+
+	//Probability there are x successful events after n total events and probability of success in an event is p
+	public static double binDistPmf(int x,int n, double p) throws IllegalArgumentException
```
Comment (dwagg) [inline Stats.java]: what is p? what is x?

File: Stats.java (hunk 9 above)
Comment (dwagg) [inline Stats.java]: nCk maybe nChooseK?

**Review by cdevan — APPROVED** [review]
Fix those things pl0x

**Comment by cdevan** [comment]
@lcc4 @dwagg Merge me!

**Comment by dwagg** [comment]
@lcc4 I'm merging my branch with the changes that we discussed in the team practice review. When we get to actually building the app we will need to decide if me merging my own branches, after review, is the flow we want to go with or if I should only merge others branches and have you merge mine. I am open to either way, just something you and I will need to establish.

**Comment by lcc4** [comment]
@dwagg Sorry, I realized I neglected to respond to this. Yes, let's discuss how to handle merging your branches - and, if I do any code, mine as well.

**Pull request merged**